pkg/jobs/canary/canary_jobs.go

```diff
@@ -269,17 +272,28 @@ func SyncCanaryJob(canary v1.Canary) error {
 		LogFail:    canary.IsTrace() || canary.IsDebug() || LogFail,
 	}
 
-	_, err := CanaryScheduler.AddJob(canary.Spec.GetSchedule(), job)
-	if err != nil {
-		return fmt.Errorf("failed to schedule canary %s/%s: %v", canary.Namespace, canary.Name, err)
+	updateTime, exists := canaryUpdateTimeCache[dbCanary.ID.String()]
+	entry := findCronEntry(canary)
+	if !exists || dbCanary.UpdatedAt.After(updateTime) || entry == nil {
+		// Remove entry if it exists
+		if entry != nil {
+			CanaryScheduler.Remove(entry.ID)
+		}
+
+		// Schedule canary for the first time
+		entryID, err := CanaryScheduler.AddJob(canary.Spec.GetSchedule(), job)
+		if err != nil {
+			return fmt.Errorf("failed to schedule canary %s/%s: %v", canary.Namespace, canary.Name, err)
+		}
+		*entry = CanaryScheduler.Entry(entryID)
```

Comment:
This panics in the first run when the `entry` var is nil.

Reply:
@yashmehrotra ^